src/test/java/org/agbrothers/englishbot/processing/processor/dictionary/AddWordDictionaryProcessorTest.java

```diff
@@ -0,0 +1,100 @@
+package org.agbrothers.englishbot.processing.processor.dictionary;
+
+import org.agbrothers.englishbot.entity.User;
+import org.agbrothers.englishbot.entity.Word;
+import org.agbrothers.englishbot.processing.ProcessingExchange;
+import org.agbrothers.englishbot.service.DictionaryService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+import static org.agbrothers.englishbot.constant.ButtonLabel.ADD_WORD;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+class AddWordDictionaryProcessorTest {
+
+    private final DictionaryService dictionaryService = mock(DictionaryService.class);
+    private final AddWordDictionaryProcessor addWordDictionaryProcessor = new AddWordDictionaryProcessor();
+
+    private User user;
+
+    @BeforeEach
+    void init() {
+        addWordDictionaryProcessor.setDictionaryService(dictionaryService);
+        user = new User();
+    }
+
+    @Test
+    void process_messageTextEqualsAddWord_EnterWord() {
+        ProcessingExchange exchange = new ProcessingExchange(user, ADD_WORD);
+
+        addWordDictionaryProcessor.process(exchange);
+
+        String result = exchange.getResponseMessageText();
+        assertEquals("Введіть слово англійською та його переклад через пробіл", result);
+        verify(dictionaryService, never()).addWord(any(Word.class));
+    }
+
+    @Test
+    void process_receiveWordAndTranslation_AddingWord() {
+        ProcessingExchange exchange1 = new ProcessingExchange(user, "sea море");
+        ProcessingExchange exchange2 = new ProcessingExchange(user, "  fly   муха ");
+
+        when(dictionaryService.getWordByEnglishValue("sea")).thenReturn(null);
+
+        addWordDictionaryProcessor.process(exchange1);
+
+        String result1 = exchange1.getResponseMessageText();
+        assertEquals("Слово sea - море було додано до словника.", result1);
+
+        ArgumentCaptor<Word> captor = ArgumentCaptor.forClass(Word.class);
+        verify(dictionaryService).addWord(captor.capture());
+        assertEquals("sea", captor.getValue().getEnglishValue());
+
+        addWordDictionaryProcessor.process(exchange2);
+        String result2 = exchange2.getResponseMessageText();
+        assertEquals("Слово fly - муха було додано до словника.", result2);
+
+        ArgumentCaptor<Word> captor1 = ArgumentCaptor.forClass(Word.class);
```

Comment:
you can use same `captor` that is on line 51

Reply:
Thanks. You're right.